scapy/layers/can.py

```diff
@@ -146,6 +149,17 @@ def extract_padding(self, p):
 bind_layers(CookedLinux, CAN, proto=12)
 
 
+class CANFD(CAN):
+    """
+    This class is used for distinction of CAN FD packets.
+    """
+    pass
+
+
+conf.l2types.register(DLT_CAN_SOCKETCAN, CANFD)
+bind_layers(CookedLinux, CANFD, proto=12)
```

Comment:
here we go... this is a major problem. You can't bind CANFD here, since CAN binds already on the same arguments... see line 149

This means... CAN and CANFD packets are not distinguishable if they come from a PCAP file. 
Have you tried Wireshark, how a CANFD packet looks like in wireshark?

Seeing this, raises the question if we should go with just one CAN(Packet) class

Reply:
sorry, proto for CANFD is 13. So major problem solved?

Reply:
Maybe... lets have some tests and see ;-D